python-mappings/pizza_menu.py

```diff
@@ -0,0 +1,57 @@
+from collections.abc import MutableMapping
+
+
+class PizzaMenu(MutableMapping):
+    def __init__(self, menu: dict):
+        self._menu = {}
+        self._first_letters = {}
+        for key, value in menu.items():
+            first_letter = key[0].lower()
+            if first_letter in self._first_letters:
+                self._raise_duplicate_key_error(key)
+            self._first_letters[first_letter] = key
+            self._menu[key] = value
+
+    def _raise_duplicate_key_error(self, key):
+        raise ValueError(
+            f"'{key}' is an invalid name. All pizzas must have unique first letters"
```

Comment:
There's a slight discrepancy between the error message here and the tutorial. Other than that, it would be nice to format the code with black.